tests/test_projectbuilder.py

```diff
@@ -570,4 +570,74 @@ def test_log(mocker, caplog, test_flit_path):
         ('INFO', 'something'),
     ]
     if sys.version_info >= (3, 8):  # stacklevel
-        assert [(record.lineno) for record in caplog.records] == [305, 305, 338, 368, 368, 562]
+        assert caplog.records[-1].lineno == 562
+
+
+def test_parse_build_system_table_type_validation():
```

Comment:
Can we parametrize this test instead?

Reply:
I would've done but it's kind of awkward when mixing test cases which raise.

Reply:
done

Reply:
The new parametrised tests fail unless I exclude `test_toml_instead_of_tomli`...